apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -56,14 +57,27 @@ pub(crate) enum HandleResponseError {
     MergeError(String),
 }
 
+impl graphql::Error {
+    fn from_connectors_runtime_error(error: &RuntimeError) -> Self {
```

Comment:
As there are multiple clonings going on here, it might be worth checking if cloning the whole object is cheaper? Also, unless we cant change `graphql::Error::builder()` functions to receive references, I would consider using only references

Reply:
i was able to remove a few clones, the builder does take references. there must be a way to remove the clone of the extensions, but because `error.code()` now returns a reference, i'm getting an "already borrowed" error.